src/libs/ajax/Metrics.ts

```diff
@@ -1,32 +1,25 @@
 import { getDefaultProperties } from '@databiosphere/bard-client';
 import _ from 'lodash/fp';
 import { authOpts, fetchBard, jsonBody } from 'src/libs/ajax/ajax-common';
-import { ensureAuthSettled } from 'src/libs/auth';
 import { getConfig } from 'src/libs/config';
 import { withErrorIgnoring } from 'src/libs/error';
 import { MetricsEventName } from 'src/libs/events';
 import * as Nav from 'src/libs/nav';
-import { authStore, getSessionId } from 'src/libs/state';
-import { v4 as uuid } from 'uuid';
+import { AuthState, authStore, getSessionId } from 'src/libs/state';
 
 export const Metrics = (signal?: AbortSignal) => {
-  const captureEventFn = async (event: MetricsEventName, details: Record<string, any> = {}): Promise<void> => {
-    await ensureAuthSettled();
-    const { signInStatus, registrationStatus } = authStore.get(); // NOTE: This is intentionally read after ensureAuthSettled
-    const isRegistered = signInStatus === 'signedIn' && registrationStatus === 'registered';
-    if (!isRegistered) {
-      authStore.update(
-        _.update('anonymousId', (id) => {
-          return id || uuid();
-        })
-      );
-    }
+  const captureEventFn = async (
+    event: MetricsEventName,
+    details: Record<string, any> = {},
+    state: AuthState = authStore.get()
+  ): Promise<void> => {
+    const isSignedIn = state.signInStatus === 'signedIn';
     const body = {
       event,
       properties: {
         ...details,
         // Users who have not registered are considered anonymous users. Send an anonymized distinct_id in that case; otherwise the user identity is captured via the auth token.
-        distinct_id: isRegistered ? undefined : authStore.get().anonymousId,
+        distinct_id: isSignedIn ? undefined : authStore.get().anonymousId,
```

Comment:
why is it undefined if they are signed in shouldnt we use their actual id?

Reply:
oh i see `otherwise the user identity is captured via the auth token.` are we sure that works how we expect?

Reply:
There might be a reason for this being registered instead of signed in. Is this a proxy for "has accepted the terms of service"? I wonder if we're avoiding sending information about them to MixPanel until they've accepted the ToS?